provider/pkg/provider/provider.go

```diff
@@ -1528,9 +1528,19 @@ func (k *kubeProvider) Create(
 	}
 	initialized, awaitErr := await.Creation(config)
 	if awaitErr != nil {
-		if req.GetPreview() && k.isDryRunDisabledError(err) {
-			logger.V(9).Infof("could not preview Create(%v): %v", urn, err)
-			return &pulumirpc.CreateResponse{Id: "", Properties: req.GetProperties()}, nil
+		if req.GetPreview() {
+			failedPreview := func() bool {
```

Comment:
What is the benefit of declaring this as a function vs. having a variable defaulted to false and set to true inside the if body?

Reply:
That would work too. I didn't think much about it, and am happy to swap approaches if that would be clearer.